consensus/polybft/state_sync_relayer.go

```diff
@@ -0,0 +1,344 @@
+	if len(sendingEvents) > 0 {
+		if err := ssr.sendTx(sendingEvents); err != nil {
+			ssr.logger.Error("failed to send tx", "block", currentBlockNumber, "events", sendingEvents, "err", err)
+		} else {
+			ssr.logger.Info("tx has been successfully sent", "block", currentBlockNumber, "events", sendingEvents)
+		}
+	}
+}
+
+func (ssr *stateSyncRelayerImpl) sendTx(events []*StateSyncRelayerEventData) error {
+	proofs := make([][]types.Hash, len(events))
+	objs := make([]*contractsapi.StateSync, len(events))
+
+	for i, event := range events {
+		proof, err := ssr.proofRetriever.GetStateSyncProof(event.EventID)
+		if err != nil {
+			return fmt.Errorf("failed to get proof for %d: %w", event.EventID, err)
+		}
+
+		// since state sync event is a map in the jsonrpc response,
+		// to not have custom logic of converting the map to state sync event
+		// json encoding is used, since it manages to successfully unmarshal the
+		// event from the marshaled map
+		raw, err := json.Marshal(proof.Metadata["StateSync"])
+		if err != nil {
+			return fmt.Errorf("failed to marshal event %d: %w", event.EventID, err)
+		}
+
+		if err = json.Unmarshal(raw, &objs[i]); err != nil {
+			return fmt.Errorf("failed to unmarshal event %d: %w", event.EventID, err)
+		}
+
+		proofs[i] = proof.Data
+	}
+
+	input, err := (&contractsapi.BatchExecuteStateReceiverFn{
+		Proofs: proofs,
+		Objs:   objs,
+	}).EncodeAbi()
+	if err != nil {
+		return err
+	}
+
+	// send batchExecute state sync
+	_, err = ssr.txRelayer.SendTransaction(&ethgo.Transaction{
+		From:  ssr.key.Address(),
+		To:    (*ethgo.Address)(&contracts.StateReceiverContract),
+		Gas:   types.StateTransactionGasLimit,
+		Input: input,
+	}, ssr.key)
+
+	return err
+}
+
+// EventSubscriber implementation
+
+// GetLogFilters returns a map of log filters for getting desired events,
+// where the key is the address of contract that emits desired events,
+// and the value is a slice of signatures of events we want to get.
+// This function is the implementation of EventSubscriber interface
+func (ssr *stateSyncRelayerImpl) GetLogFilters() map[types.Address][]types.Hash {
+	var (
+		stateSyncResultEvent contractsapi.StateSyncResultEvent
+		newCommitmentEvent   contractsapi.NewCommitmentEvent
+	)
+
+	return map[types.Address][]types.Hash{
+		contracts.StateReceiverContract: {
+			types.Hash(stateSyncResultEvent.Sig()),
+			types.Hash(newCommitmentEvent.Sig()),
+		},
+	}
+}
+
+// ProcessLog is the implementation of EventSubscriber interface,
+// used to handle a log defined in GetLogFilters, provided by event provider
+func (ssr *stateSyncRelayerImpl) ProcessLog(header *types.Header, log *ethgo.Log, dbTx *bolt.Tx) error {
+	var (
+		commitEvent          contractsapi.NewCommitmentEvent
+		stateSyncResultEvent contractsapi.StateSyncResultEvent
+	)
+
+	switch log.Topics[0] {
+	case commitEvent.Sig():
+		_, err := commitEvent.ParseLog(log)
+		if err != nil {
+			return err
+		}
+
+		newEvents := []*StateSyncRelayerEventData{}
+
```

Comment:
Initialize with `make` and provide capacity?

```suggestion
		newEvents := make([]*StateSyncRelayerEventData, 0, commitEvent.StartID.Uint64()-commitEvent.StartID.Uint64()+1)
```

Reply:
fixed https://github.com/0xPolygon/polygon-edge/pull/1899/commits/648a4f5c02d020891a5e3f1d3c9be482dc8c8639